rest/test_results_routes.go

```diff
@@ -26,6 +27,146 @@ const (
 	testResultsBaseTaskID = "base_task_id"
 )
 
+type testResultsTasksBaseHandler struct {
+	sc      data.Connector
+	payload struct {
+		TaskOpts   []data.TestResultsTaskOptions         `json:"tasks"`
+		FilterOpts *data.TestResultsFilterAndSortOptions `json:"filters"`
+	}
+}
+
+func (h *testResultsTasksBaseHandler) Parse(_ context.Context, r *http.Request) error {
+	if r.Body == nil {
+		return errors.New("missing request payload")
+	}
+	body := utility.NewRequestReader(r)
+	defer body.Close()
+
+	if err := json.NewDecoder(r.Body).Decode(&h.payload); err != nil {
+		return errors.Wrap(err, "decoding JSON request payload")
+	}
+
+	if len(h.payload.TaskOpts) == 0 {
+		return errors.New("must specify at least one task in the request payload")
+	}
+
+	return nil
+}
+
+///////////////////////////////////////////////////////////////////////////////
+//
+// GET /test_results/tasks
+
+type testResultsGetByTasksHandler struct {
+	testResultsTasksBaseHandler
+}
+
+func makeGetTestResultsByTasks(sc data.Connector) *testResultsGetByTasksHandler {
+	h := &testResultsGetByTasksHandler{}
+	h.sc = sc
+
+	return h
+}
+
+func (h *testResultsGetByTasksHandler) Factory() gimlet.RouteHandler {
+	newHandler := &testResultsGetByTasksHandler{}
+	newHandler.sc = h.sc
+
+	return newHandler
+}
+
+func (h *testResultsGetByTasksHandler) Run(ctx context.Context) gimlet.Responder {
+	testResults, err := h.sc.FindTestResults(ctx, h.payload.TaskOpts, h.payload.FilterOpts)
+	if err != nil {
+		err = errors.Wrap(err, "getting test results by tasks")
+		logFindError(err, message.Fields{
+			"request":         gimlet.GetRequestID(ctx),
+			"method":          "GET",
+			"route":           "/test_results/tasks",
+			"request_payload": h.payload,
+		})
+		return gimlet.MakeJSONErrorResponder(err)
+	}
+
+	return gimlet.NewJSONResponse(testResults)
```

Comment:
is it necessary to set `pages` on the response?

Reply:
Not anymore: since we are now taking in a request payload (which isn't really what GET requests should do) the pagination params are part of the request body rather than the query params. The `pages` only support setting the query parameters, we would also need to set the body. After speaking with @khelif96 this isn't a problem since the Evergreen code that talks to Cedar takes care of pagination itself via the graphql queries rather than using the `next` header from the response.

Reply:
ah, I understand